td-shim/src/event_log.rs

```diff
@@ -16,10 +19,143 @@ pub const PCR_DIGEST_NUM: usize = 1;
 pub const CC_EVENT_HEADER_SIZE: usize = 66;
 pub const CCEL_CC_TYPE_TDX: u8 = 2;
 
+pub const EV_NO_ACTION: u32 = 0x00000003;
+pub const EV_SEPARATOR: u32 = 0x00000004;
+pub const EV_PLATFORM_CONFIG_FLAGS: u32 = 0x0000000A;
 pub const EV_EFI_EVENT_BASE: u32 = 0x80000000;
-pub const EV_PLATFORM_CONFIG_FLAGS: u32 = EV_EFI_EVENT_BASE + 0x0000000A;
+pub const EV_EFI_PLATFORM_FIRMWARE_BLOB2: u32 = EV_EFI_EVENT_BASE + 0xA;
 pub const EV_EFI_HANDOFF_TABLES2: u32 = EV_EFI_EVENT_BASE + 0xB;
 
+pub const PLATFORM_CONFIG_HOB: &[u8] = b"td_hob\0";
+pub const PLATFORM_CONFIG_PAYLOAD_PARAMETER: &[u8] = b"td_payload_info\0";
+pub const PLATFORM_CONFIG_SECURE_POLICY_DB: &[u8] = b"secure_policy_db";
+pub const PLATFORM_CONFIG_SECURE_AUTHORITY: &[u8] = b"secure_authority";
+pub const PLATFORM_CONFIG_SVN: &[u8] = b"td_payload_svn\0";
+pub const PLATFORM_FIRMWARE_BLOB2_PAYLOAD: &[u8] = b"td_payload\0";
+
+const VENDOR_INFO_SIZE: usize = 7;
+const VENDOR_INFO: &[u8; VENDOR_INFO_SIZE] = b"td_shim";
+
+/// TCG_EfiSpecIdEvent is the first event in the event log
+/// It is used to determine the version and format of the events in the log, identify
+/// the number and size of the recorded digests
+///
+/// Defined in TCG PC Client Platform Firmware Profile Specification:
+/// 'Table 20 TCG_EfiSpecIdEvent'
+#[repr(C)]
+pub struct TcgEfiSpecIdevent {
+    pub signature: [u8; 16],
+    pub platform_class: u32,
+    pub spec_version_minor: u8,
+    pub spec_version_major: u8,
+    pub spec_errata: u8,
+    pub uintn_size: u8,
+    pub number_of_algorithms: u32,
+    pub digest_sizes: [TcgEfiSpecIdEventAlgorithmSize; 1],
+    pub vendor_info_size: u8,
+    // Fix the vendor info size to VENDOR_INFO_SIZE
+    pub vendor_info: [u8; VENDOR_INFO_SIZE],
+}
+
+#[repr(C)]
+pub struct TcgEfiSpecIdEventAlgorithmSize {
+    algorithm_id: u16,
+    digest_size: u16,
+}
+
+impl TcgEfiSpecIdevent {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        unsafe { &*slice_from_raw_parts(self as *const Self as *const u8, size_of::<Self>()) }
+    }
+}
+
+impl Default for TcgEfiSpecIdevent {
+    fn default() -> Self {
+        Self {
+            signature: *b"Spec ID Event01\0",
```

Comment:
Why you use "01", the PFP spec says "03", refer to Table 20 TCG_EfiSpecIdEvent.

Reply:
Yes it should be '03', misunderstood the content and thought it is an example